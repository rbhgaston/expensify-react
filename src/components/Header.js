import {NavLink} from 'react-router-dom';
import React from 'react';
import {startLogout} from '../actions/auth';
import {connect} from 'react-redux'

export const Header = (props) =>  {
    return (
        <header>
            <h1>Expensify</h1>
            <NavLink exact to='/dashboard' activeClassName='is-active'>Dashboard</NavLink>
            <NavLink exact to='/create' activeClassName='is-active'>Create Expense</NavLink>
            <button onClick={props.startLogout}>Logout</button>
        </header>
    )
}

const mapDispatchToProps = dispatch => ({
    startLogout: () => dispatch(startLogout())
})

export default connect(undefined, mapDispatchToProps)(Header)