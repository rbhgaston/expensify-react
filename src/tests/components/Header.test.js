import React from 'react'
import Header from '../../components/Header';
// import ReactShallowRenderer from 'react-test-renderer/shallow'
import {shallow} from 'enzyme'

test('render Header component', () => {
    const wrapper = shallow(<Header/>)
    expect(wrapper).toMatchSnapshot()
    // const renderer = new ReactShallowRenderer()
    // renderer.render(<Header/>)
    // expect(renderer.getRenderOutput()).toMatchSnapshot()
})