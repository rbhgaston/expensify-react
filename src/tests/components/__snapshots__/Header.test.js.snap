// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Header component 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    exact={true}
    to="/dashboard"
  >
    Dashboard
  </NavLink>
  <NavLink
    activeClassName="is-active"
    exact={true}
    to="/create"
  >
    Create Expense
  </NavLink>
  <button
    onClick={[Function]}
  >
    Logout
  </button>
</header>
`;
